app/controllers/api/v1/users_controller.rb

```diff
@@ -1,22 +1,26 @@
 class Api::V1::UsersController < ApplicationController
+  before_action :authorized, only: %i[auto_login]
+
   # REGISTER
   def create
     @user = User.create(user_params)
     if @user.valid?
-      # token = encode_token({user_id: @user.id})
-      render json: { user: @user }
+      token = encode_token({ user_id: @user.id })
+      render json: { user: @user, token: }
     else
       render json: { error: 'Invalid username or password' }
     end
   end
 
   # LOGGING IN
   def login
+    return unless @user.save
+
     @user = User.find_by(username: params[:username])
 
     if @user&.authenticate(params[:password])
-      # token = encode_token({user_id: @user.id})
-      render json: { user: @user }
+      token = encode_token({ user_id: @user.id })
+      render json: { user: @user, token: }
     else
       render json: { error: 'Invalid username or password' }
```

Comment:
- [ ] When you try to log in if the user is not registered please try to register the user. You have done the authentication but try to link the create method if the user not registered.
  - As we agreed we don't need `/signup` API is so we need the `/login` API to Perform create a user
  
change the else block with create user https://github.com/oluyaratosin123/Final-Capstone-Back-End/blob/b2f82ec6d812b086b34864e0e9453f57eb7cdc76/app/controllers/api/v1/users_controller.rb#L16

```ruby
 def login
    return unless @user.save

    @user = User.find_by(username: params[:username])

    if @user&.authenticate(params[:password])
      token = encode_token({ user_id: @user.id })
      render json: { user: @user, token: }
    else
   #create user here try to send token and the user registered after saving
    end
  end
```